Example-CustomUserGroups-REXX/USRGRP01.rexx

```diff
@@ -0,0 +1,3123 @@
+ *   targetName  - The target name of the object to send the request to
+ *   requestBody - The JSON request body
+ *   client_corr - The client correlator
+ *   encoding    - The encoding as an integer
+ *   timeout     - The timeout as an integer
+ *
+ * Output: On success, returns the respones body. On failure, returns
+ *         an empty string
+ *
+ * Side-Effects: None
+ *
+ **********************************************************************/
+GetRequest:
+
+  GET_URI_DEFAULT = '/api/cpcs'
+  GET_TARGETNAME_DEFAULT = ''
+  GET_REQUESTBODY_DEFAULT = ''
+  GET_CLIENTCORR_DEFAULT = ''
+  GET_ENCODING_DEFAULT = 0
+  /* use default timeout of 60 minutes */
+  GET_TIMEOUT_DEFAULT = 0
+
+  uriArg = Arg(1)
+  targetnameArg = Arg(2)
+  requestBodyArg = Arg(3)
+  clientCorrArg = Arg(4)
+  encodingArg = Arg(5)
+  timeoutArg = Arg(6)
+
+  /*
+   * Reset the STEM variables
+   */
+  Drop userRequest.
+  Drop response.
+
+  /*
+   * Set the default values for the GET request
+   */
+
+  specifiedURI = GET_URI_DEFAULT
+  specifiedTargetName = GET_TARGETNAME_DEFAULT
+  specifiedRequestBody = GET_REQUESTBODY_DEFAULT
+  specifiedClientCorr = GET_CLIENTCORR_DEFAULT
+  specifiedEncoding = GET_ENCODING_DEFAULT
+  specifiedTimeout = GET_TIMEOUT_DEFAULT
+
+  /*
+   * Assign the fields appropriately
+   */
+  If uriArg <> '' Then
+    specifiedURI = uriArg
+
+  If targetnameArg <> '' Then
+    specifiedTargetName = targetnameArg
+
+  If requestBodyArg <> '' Then
+    specifiedRequestBody = requestBodyArg
+
+  If clientCorrArg <> '' Then
+    specifiedClientCorr = clientCorrArg
+
+  If encodingArg <> 0 & encodingArg <> '' Then
+    specifiedEncoding = encodingArg
+
+  If timeoutArg <> 0 & timeoutArg <> '' Then
+    specifiedTimeout = timeoutArg
+
+  userRequest.HTTPMETHOD = HWI_REST_GET
+  userRequest.URI = specifiedURI
+  userRequest.TARGETNAME = specifiedTargetName
+  userRequest.REQUESTBODY = specifiedRequestBody
+  userRequest.CLIENTCORRELATOR = specifiedClientCorr
+  userRequest.ENCODING = specifiedEncoding
+  userRequest.REQUESTTIMEOUT = specifiedTimeout
+
+  Say
+  Say '------->'
+  Say 'GET request being made....'
+  Say 'URI: ' || userRequest.URI
+
+  If userRequest.TARGETNAME <> '' Then
+    Say 'TARGETNAME: '||userRequest.TARGETNAME
+
+  If userRequest.REQUESTBODY <> '' Then
+    Say 'REQUEST BODY: '||userRequest.REQUESTBODY
+
+  If userRequest.CLIENTCORRELATOR <> '' Then
+    Say 'CLIENT CORRELATOR: '||userRequest.CLIENTCORRELATOR
+
+  If userRequest.ENCODING <> '' Then
+    Say 'ENCODING: '||userRequest.ENCODING
+
+  If userRequest.REQUESTTIMEOUT <> '' Then
+    Say 'TIMEOUT: '||userRequest.REQUESTTIMEOUT
+
+  Address BCPII "HWIREST userRequest. response."
+
+  /* GET requests return 200 when successful */
+
+  Call SurfaceResponse RC
+  Say
+  Say '<-------'
+  Say
+
+  responseData = ''
+
+  If RC = 0 & response.httpStatus = 200 Then
+    responseData = response.responseBody
+
+  Drop userRequest.
+  Drop response.

```

Comment:
just curious why?

Reply:
I was paranoid someone might modify the REXX and accidentally forget to drop the response and user request stem variables before reusing them